haskell/private/haskell_impl.bzl

```diff
@@ -95,19 +129,43 @@ def haskell_library_impl(ctx):
     cc,
     java,
     dep_info,
-    cpp_defines = ctx.file._cpp_defines,
-    compiler_flags = ctx.attr.compiler_flags,
     srcs = ctx.files.srcs,
     extra_srcs = depset(ctx.files.extra_srcs),
+    cpp_defines = ctx.file._cpp_defines,
+    compiler_flags = ctx.attr.compiler_flags,
+    with_profiling = False,
     my_pkg_id = my_pkg_id,
   )
 
+  c_p = None
+
+  if with_profiling:
+    c_p = hs.toolchain.actions.compile_library(
+      hs,
+      cc,
+      java,
+      dep_info,
+      srcs = ctx.files.srcs,
+      # NOTE We must make the object files compiled without profiling
```

Comment:
Or use `-fext-external-interpreter`, which I believe should resolve this kind of problem. See https://ghc.haskell.org/trac/ghc/wiki/RemoteGHCi. I'm surprised it's still not the default - we should pass that flag systematically if not.

Reply:
You meant `-fexternal-interpreter` I think?

https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/ghci.html?highlight=#ghc-flag--fexternal-interpreter

This looks good, although I'm not sure it makes sense to spend more time switching to that as I've already spent more hours than I expected to spend with this. Also, right now we're doing what Cabal is doing here. Of course Cabal works that way because it needs to support older GHC, but maybe there are other undiscovered issues with `-fexternal-interpreter`? CC @facundominguez, he mentioned something like "there may be issues with that".

Let's open an issue for the future and leave this "as is" so I can spend more time on client's stuff instead of re-working the working solution that replicates what Cabal does?